### ISSUE TITLE
10bit usage

### ISSUE THREAD

**Issue by hohMiyazawa** [Calve33/Skoll3webm#4]
Just to document this. The sane options would be:
- 10bit always
- 10bit when 10bit input

10bit encoding from 8bit sources gives some coding gain, but nothing impressive.

**Comment by Calve33** [comment]
>10bit encoding from 8bit sources gives some coding gain, but nothing impressive.

Yeah. I remeber it being the way anime groups squezing debanding into 8 bit source without overbloating. ALbeit ithere is some paper that claims that 10 bit (at least on h264 and vp9) just improves efficiency without further manipulations

**Comment by hohMiyazawa** [comment]
Resizing an 8bit source can also increase the effective bit depth slightly.

But taking advantage of this likely requires rewriting the video scaling portion of the script. By default, ffmpeg uses the same pixel format for the input and output of the scaling.

**Comment by hohMiyazawa** [comment]
The default behaviour is right now to do "10bit when 10bit input". That's at least better than what it does for libvpx

**Comment by Calve33** [comment]
>That's at least better than what it does for libvpx

Dunno about that. VP9 10bit non-HDR playback in browsers is _really_ untested grounds, unlike AV1 10bit

**Comment by hohMiyazawa** [comment]
What I mean is that mpv-webm never made 10bit encodes before, since libvpx automatically used yuv420p (even if the input was yuv420p10le) with the settings in the script. libaom on the other hand automatically preserves the pixel format.

So before, you never got 10bit encodes, and now with this AV1 patch you get 10bit encodes when they matter the most.

But I agree that I have no idea how 10bit non-HDR behaves. Probably not worth implementing.

**Comment by Calve33** [comment]
> mpv-webm never made 10bit encodes before

In case of using x264 through mpv-webm it does encode 10 bit source into 10 bit clip (and it is really annoying behavior when web sharing is primary use). But for libvpx hacknyed implementation of 10 bit on HDR _and_ 10bit AV1 playback is a standard for hardware playback it does not matter

**Comment by hohMiyazawa** [comment]
As you can see, I have never made h264 encodes for the web :)

**Comment by hohMiyazawa** [comment]
Is that something that should be changed too?
10bit h264 sounds actively bad for a quick and convenient clip script. The use cases for 10bit h264 aren't overlapping with this at all.

**Comment by Calve33** [comment]
>Is that something that should be changed too?

That would be really neat since copy-pasta       "--ovcopts-add=format=yuv420p" do not work
